jax/_src/ops/special.py

```diff
@@ -95,3 +97,22 @@ def logsumexp(a: ArrayLike, axis: Axis = None, b: ArrayLike | None = None,
     with jax.debug_nans(False):
       out = jnp.where(sign < 0, jnp.array(np.nan, dtype=out.dtype), out)
   return out
+
+@logsumexp.defjvp
+def _logsumexp_jvp(axis, keepdims, return_sign, where, primals, tangents):
+  a, b = primals
+  a_dot, b_dot = tangents
+  out = logsumexp(a, axis, b, keepdims, return_sign, where)
+
+  if return_sign:
+    out, sign = out
+
+  if b is None:
+    b = jnp.ones_like(a)
+  out_dot = jnp.exp(
+      logsumexp(a, axis, b * a_dot + b_dot, keepdims, where) - out)
+
+  if return_sign:
+    return (out, sign), out_dot
```

Comment:
I think this probably needs a `sign_dot` here too. I think the tangents always need to have the same pytree structure as the primals.

Reply:
I could be wrong, but I think `sign_dot` is included in `out_dot` if `return_sign` is True.

Reply:
Added a test with `return_sign=True`.